hls/lib/hls_endpoint.ex

```diff
@@ -259,8 +263,18 @@ defmodule Membrane.RTC.Engine.Endpoint.HLS do
   end
 
   @impl true
-  def handle_parent_notification({:new_tracks, _tracks}, _ctx, %{subscribe_mode: :manual} = state) do
-    {[], state}
+  def handle_parent_notification({:new_tracks, tracks}, ctx, %{subscribe_mode: :manual} = state) do
+    subscribed_tracks = Map.keys(state.tracks)
+
+    subscribed_tracks =
+      tracks
+      |> Enum.filter(fn track ->
+        track.origin in state.subscribed_endpoints and track.id not in subscribed_tracks
```

Comment:
This `and` shouldn't be required

```suggestion
        track.origin in state.subscribed_endpoints
```